### PR TITLE
sv_pure bypass based on @notgoodusername #1802 pull request, updated triggerbot hitchance

### PR DESCRIPTION
**Pull request by wemanzoz**
sv_pure bypass
I finally did it, code is pasted from [@notgoodusername](https://github.com/notgoodusename)'s [osiriswithstuff](https://github.com/notgoodusename/osiriswithstuff) repo

special thanks to [@VermelhoOFI](https://www.unknowncheats.me/forum/members/2002919.html) on unknowncheats for helping me

triggerbot hitchance
thanks [@pubsense](https://github.com/pubsense) for [#1129](https://github.com/danielkrupinski/Osiris/pull/1129)

Vicarious#0 0 0 2 on discord

### CODE REVIEW

**Comment by wemanzoz** [comment]
ps: i dont the the ImGui thing in gui.cpp because in my build, i want it always on

File: Osiris/Hooks.h [hunk 1]
```diff
@@ -55,6 +55,8 @@ class Hooks {
     HookType sound;
     HookType surface;
     HookType viewRender;
+    VmtSwap networkChannel;
```
Comment (mkot2) [inline Osiris/Hooks.h:58]: change to hooktype, vmt is probably detected and there is no reason not to use hooktype

Comment (wemanzoz) [reply Osiris/Hooks.h:58]: i wish i can lol, codes pasted

Comment (wemanzoz) [reply Osiris/Hooks.h:58]: any1 wanna do it for me?

**Comment by Kruziikrel1** [comment]
Thank you this is working but I need Allow loose File loads

File: Osiris/SDK/NetworkChannel.h [hunk 2]
```diff
@@ -7,5 +7,16 @@ class NetworkChannel {
     VIRTUAL_METHOD(float, getLatency, 9, (int flow), (this, flow))
 
     std::byte pad[44];
+    int OutSequenceNr;
+    int InSequenceNr;
+    int OutSequenceNrAck;
+    int OutReliableState;
+    int InReliableState;
```
Comment (dumbasPL) [inline Osiris/SDK/NetworkChannel.h:14]: nice paste XD  
If you want to add 5 int's here you have to deduct `5*sizeof(int)` from the padding above.  
Have you even tested this before making a pr?

Comment (wemanzoz) [reply Osiris/SDK/NetworkChannel.h:14]: i know its a paste LOL do you think ill waste my time and selfcode it when theres a source with it? its literally called "based on" so i literally pasted it you dumb fuck, and i wont go to uc and ask anyone if im about to fucking paste you dumb fucking moron. what a monkey

Comment (mkot2) [reply Osiris/SDK/NetworkChannel.h:14]: lmao what a joke

File: Osiris/Hooks.cpp [hunk 3]
```diff
@@ -143,6 +144,16 @@ static HRESULT __stdcall reset(IDirect3DDevice9* device, D3DPRESENT_PARAMETERS*
 
 #endif
 
+static bool __fastcall sendNetMsg(void* networkchannel, void* edx, NetworkMessage& msg, bool bForceReliable, bool bVoice)
+{
+    auto original = hooks->networkChannel.getOriginal<bool, NetworkMessage&, bool, bool>(40, msg, bForceReliable, bVoice);
```
Comment (mkot2) [inline Osiris/Hooks.cpp:149]: `auto original = hooks->networkChannel.getOriginal<bool, 40, NetworkMessage&, bool, bool>(msg, bForceReliable, bVoice);`
lmao did you even test this

File: Osiris/SDK/NetworkChannel.h (hunk 2 above)
Comment (wemanzoz) [reply Osiris/SDK/NetworkChannel.h:14]: it actually works lmfao

File: Osiris/Hooks.cpp (hunk 3 above)
Comment (wemanzoz) [reply Osiris/Hooks.cpp:149]: yeah whats up, it worked. dont want to use it? fine i dont care, its up to you

Comment (wemanzoz) [reply Osiris/Hooks.cpp:149]: > `auto original = hooks->networkChannel.getOriginal<bool, 40, NetworkMessage&, bool, bool>(msg, bForceReliable, bVoice);`
> lmao did you even test this

yeah whats up, it worked. dont want to use it? fine i dont care, its up to you

Comment (wemanzoz) [reply Osiris/Hooks.cpp:149]: if you dont want to usse this pull, then fuck off will ya?

File: Osiris/SDK/NetworkChannel.h (hunk 2 above)
Comment (Bakedpurple) [reply Osiris/SDK/NetworkChannel.h:14]: These monkeys keep complaining on a clearly pasted pr but won't make a new pr on a better sv_pure bypass yikes

Comment (wemanzoz) [reply Osiris/SDK/NetworkChannel.h:14]: > These monkeys keep complaining on a clearly pasted pr but won't make a new pr on a better sv_pure bypass yikes

true

**Comment by cristeigabriel** [comment]
Poor implementation. Consider:
- Not using VMT for this
- If you can't do the first, only do this in LevelInit (xrefs: "LevelInit" for primary handler, 	#STR: "cl_predict 1", "cl_predict 0", "(mapname)" for pre-entity, and post-entity is right above. They're inlined, BTW, because the NOINLINE specification is not defined in live builds)
- Explain this better. What this actually does is prevent the processing of a shared module protobuf.

**Comment by cristeigabriel** [comment]
engine.dll SendNetMsg xrefs:
        // #STR: "Buffer overflow in send net message", "SendNetMsg %s: stream[%s] buffer overflow
        // (maxsize = %d)!

**Pull request closed without merging**